### PR TITLE
Route Admin Router error log to /dev/log

### PR DESCRIPTION
**Pull request by jongiddy**
## High-level description

Route error_log messages to /dev/log instead of stderr.

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [https://jira.mesosphere.com/browse/DCOS-43956](https://jira.mesosphere.com/browse/DCOS-43956) Switch AR `error_log` directive to log using `syslog` protocol

## Related tickets (optional)

Other tickets related to this change:

  - [DCOS_OSS-<number>](https://jira.mesosphere.com/browse/DCOS_OSS-<number>) Foo the Bar so it stops Bazzing.

## Checklist for all PRs

  - [x] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change:
  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by jgehrcke** [comment]
I re-ran the build job; it's a requirement for the other jobs to run (e.g. teamcity/dcos/test/dcos-docker/static )

**Comment by gpaul** [comment]
Looks like the adminrouter tests need updating to account for the logging changes:
```
17:25:35 E           AssertionError: assert {'Invalid token': 1} == {}
17:25:35 E             Left contains more items:
17:25:35 E             {'Invalid token': 1}
17:25:35 E             Full diff:
17:25:35 E             - {'Invalid token': 1}
17:25:35 E             + {}
```

**Comment by jongiddy** [comment]
I need to use a deepcopy to copy the SearchCriteria in the `assert_error_log` dictionary.

**Comment by jongiddy** [comment]
@mesosphere-mergebot bump-ee

**Comment by jongiddy** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by jongiddy** [comment]
@vespian @ahairape @jgehrcke @mhrabovcin I'd appreciate a review of this change.

File: packages/adminrouter/extra/src/includes/main/common.conf [hunk 1]
```diff
@@ -3,7 +3,11 @@
 # * some of the log messages that are required by test
 #   harness are not important enough to justify anything
 #   higher than info level.
-error_log stderr info;
+# Use a non-blocking Unix domain socket. If journald fails to read
+# the socket quickly enough, log messages may be lost, but this is
+# better than logging to a blocking connection (e.g. stderr) which
+# may cause Admin Router to hang.
```
Comment (jgehrcke) [inline packages/adminrouter/extra/src/includes/main/common.conf]: Thanks for the code comment!

File: packages/adminrouter/extra/src/includes/main/common.conf [hunk 2]
```diff
@@ -3,7 +3,11 @@
...
+error_log syslog:server=unix:/dev/log info;
```
Comment (jgehrcke) [inline packages/adminrouter/extra/src/includes/main/common.conf]: Did you talk to @timaa2k about the syslog config, like about setting a specific facility?

File: packages/adminrouter/extra/src/test-harness/modules/runner/common.py [hunk 3]
```diff
@@ -678,7 +678,9 @@ def _set_ar_cmdline(self):
 
         self._args = [self.binary,
                       '-c', self.config_path,
-                      '-g', 'daemon off;',
+                      # For tests, NGINX must run in foreground and direct
+                      # error_log to stderr.
+                      '-g', 'daemon off;error_log stderr info;',
```
Comment (jgehrcke) [inline packages/adminrouter/extra/src/test-harness/modules/runner/common.py]: :+1:

File: packages/adminrouter/extra/src/test-harness/modules/runner/common.py [hunk 4]
```diff
@@ -809,6 +813,11 @@ def make_url_from_path(self, path='/exhibitor/some/path'):
 
         return base + path[1:]
 
+    @property
+    def syslog_line_buffer(self):
+        """Return line buffer where all syslog output of this object resides"""
+        return self._syslog_mock.stdout_line_buffer
```
Comment (jgehrcke) [inline packages/adminrouter/extra/src/test-harness/modules/runner/common.py]: Thanks for fighting the Admin Router test harness! We keep it green and make @vespian proud :)

File: packages/adminrouter/extra/src/includes/main/common.conf (hunk 2 above)
Comment (ahairape) [reply packages/adminrouter/extra/src/includes/main/common.conf]: We are setting `access_log` in `extra/src/includes/http/common.conf` this way:
`access_log syslog:server=unix:/dev/log,facility=daemon customformat;`

Any reason not to set the `error_log` similarly? 
CC: @timaa2k

Comment (jongiddy) [reply packages/adminrouter/extra/src/includes/main/common.conf]: D'oh! I even noted this in the JIRA issue, but then forgot to add it to the code.

**Comment by jongiddy** [comment]
@mesosphere-mergebot bump-ee

**Comment by jongiddy** [comment]
@mesosphere-mergebot bump-ee

**Comment by jongiddy** [comment]
Override please:
```
@mesosphere-mergebot override-status teamcity/dcos/test/aws/onprem/static/group2 https://jira.mesosphere.com/browse/DCOS_OSS-4568
```

**Comment by jgehrcke** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/aws/onprem/static/group2 https://jira.mesosphere.com/browse/DCOS_OSS-4568

**Pull request merged**